Platform/AlderlakeBoardPkg/Library/Stage1ABoardInitLib/Stage1ABoardInitLib.c

```diff
@@ -181,6 +272,10 @@ BoardInit (
       }
       AsmWriteMsr64(MsrIdx, (SIZE_4GB - AdjLen) | CACHE_WRITEPROTECTED);
       AsmWriteMsr64(MsrIdx + 1, (MskLen - AdjLen) | B_CACHE_MTRR_VALID);
+    } else {
+      if (PcdGetBool (PcdFastBootEnabled)) {
+        ExpandCacheAsRam ();
```

Comment:
This new function is quite complicated and it might take several MTRRs for MRC data and I didn't see it is freed.
Since other BIOS region also need be set to cacheable, could you have a try to set a free MTRR MSR to cover whole BIOS region (similar with BtG disabled case)? Hopefully it could also solve the boot performance issue, and use less MSR and benefit other bios component.

Reply:
1. TempRamExit will free all used var MTRRs after PreMem stage. Usually there are 10 var MTRRs, and 2-3 will be used when ACM initializes NEM (based on the size of IBB). I didn’t see var MTRR used elsewhere in Stage1A.
2. This is just to make the code more general and avoid not having enough LLC to cover the entire BIOS region on some SKU. I don't have enough resources to test all SKUs so I don't know if they have enough LLC.

@jwu55 Hi Ji, do you see any risk about 1 single MTRR?

Reply:
ok. Then it is not an issue to use multiple MTRRs.  
I know there are some limitations on the NEM size (e.g. NEM size = stack data region + code data region; Effective cache size = NEM size + reserved reserved region; both stack data region and code region should be continuous region, etc.).  I don't know the exact reasons for these limitations. I think it should work since a) it works in BtG disabled  case, b) MRC data region is a new code region which is not a continuous region with IBB, it already breaks a limitation.

Reply:
I think you are right. Since it works fine with BTG disabled, it should work the same with BtG enabled. My worries about Cache size are unnecessary.
I updated patch please check.
MTRR value of BtG enabled case as below:
1. Before change:
MTRR: ID(0) BASE(FEF00006) Mask(FFF00800) 
MTRR: ID(1) BASE(FFE00005) Mask(FFF00800) (Stage1A + ACM0)
MTRR: ID(2) BASE(FFFE0005) Mask(FFFE0800) (Stage1B)

2. After
MTRR: ID(0) BASE(FEF00006) Mask(FFF00800) 
MTRR: ID(1) BASE(FF000005) Mask(FF000800) (Full flash region same as BtG disabled case)

Reply:
with this new change, did you see any boot performance impact comparing with your previous change (only add MRC data region)? 
BTW, to make code simple, without clearing other MTRRs should be OK. The hardware should be able to support the overlap.

Reply:
> with this new change, did you see any boot performance impact comparing with your previous change (only add MRC data region)?
No, performance is same as previous change (FSPM reduced 100ms and no other impact).

> BTW, to make code simple, without clearing other MTRRs should be OK. The hardware should be able to support the overlap.
Sounds good, then we just need to add fastboot check and no need to add new function. I tested and it works fine, patch updated.